src/TorchSharp/Tensor/Tensor.cs

```diff
@@ -5142,6 +5142,64 @@ public IList<Tensor> nonzero_as_list()
                 return t.chunk(t.shape[1], dim: 1);
             }
 
+            [DllImport("LibTorchSharp")]
+            extern static IntPtr THSTensor_roll(IntPtr tensor, IntPtr shifts, int shLength, IntPtr dims, long dimLength);
+
+            /// <summary>
+            /// Roll the tensor along the given dimension(s).
+            /// Elements that are shifted beyond the last position are re-introduced at the first position.
+            /// If a dimension is not specified, the tensor will be flattened before rolling and then restored to the original shape.
+            /// </summary>
+            public Tensor roll(long shifts, long? dims = null)
+            {
+                if (dims.HasValue) {
+                    return roll(new long[] { shifts }, new long[] { dims.Value });
```

Comment:
>new long[]  [](http://example.com/codeflow?start=32&length=11)

Thanks for making Roll take a span. now you can convert all these long array allocations to stackalloc's.

Reply:
Hmm... When I tried changing to ReadOnlySpan for a few other APIs (torch.zeros, for example), that seems to not work with F#.

@dsyme -- this doesn't work if torch.zeros takes a ReadOnlySpan:

```F#
let mutable pe = torch.zeros([| maxLen; dmodel|])
```

Do I need to overload ROS<long> and IEnumerable<long>?

Reply:
>> Do I need to overload ROS and IEnumerable?

It seems I need to have roll(ReadOnlySpan<long>,...) and roll(long[],...) as well as a private _roll(ReadOnlySpan<long>...) with the actual implementation.